Add quick links to Administration and Moderation Reports from Web UI

ThisIsMissEm opened the pull request:
Inspired by #23747, whilst showing the number of reports and the other "dashboard" statistics would be complicated, we can do a simple change that at least makes getting to either the Administration or Moderation Reports sections easier.

For an Owner, the following is what will now show in the sidebar in standard view: 
<img width="305" alt="Screenshot 2023-05-04 at 8 17 15 am" src="https://user-images.githubusercontent.com/30827/236125360-0b06965a-8444-463d-a578-a66c3657ae25.png">

and advanced view:
<img width="342" alt="Screenshot 2023-05-04 at 8 26 30 am" src="https://user-images.githubusercontent.com/30827/236126719-f53c1056-dab7-44d2-b3aa-1859d497193c.png">

I took the approach of adding helper methods for checking the permissions, as I didn't what to duplicate the LSB check in multiple places; we could eventually change this to a more generalised form `can(permissions, "manage_reports")` and have that powered by the logic that exists in UserRoles (but this would require exporting a map of flag name -> flag value for javascript to read, and you'd then want typescript types and all the rest).

It may make sense to change `permissions.ts` to use the same bit-shifting syntax though, so `1 << 3` for `0x0000000000000008` which might at least make it clearer that the values are the same as those in UserRoles

ThisIsMissEm committed:
Add quick links to Administration and Moderation Reports from Web UI

ClearlyClaire reviewed (commented):
This is ok from a technical standpoint, but I'm afraid this makes the barrage of sidebar options even worse.

I'm tagging @Gargron as this is a UI change that probably needs to go through him.

vmstan commented:
I would love to see the Administration link for folks with access more prominent.

ThisIsMissEm commented:
Have rebased this in case you still wish to consider it. I think it won't make the barrage of sidebar options that much worse, given it only affects moderators & admins.

ThisIsMissEm commented on app/javascript/mastodon/locales/en.json:
```diff
@@ -413,6 +414,7 @@
   "navigation_bar.follows_and_followers": "Follows and followers",
   "navigation_bar.lists": "Lists",
   "navigation_bar.logout": "Logout",
+  "navigation_bar.moderation": "Moderation Reports",
```
arguably this should be called `moderation_reports` instead of `moderation`...

ThisIsMissEm commented:
@Gargron please let me know if I should rebase this PR.

ThisIsMissEm commented:
@mjankowski I'm no longer sure how to rebase this branch after the icon changes, I just get: 
<img width="358" alt="Screenshot 2024-07-16 at 20 38 53" src="https://github.com/user-attachments/assets/ba1a1084-1752-4114-8d3c-1b71e6bdf6b7">

I've pushed what I think the changes needed to be, but as you can see it doesn't work.

ThisIsMissEm commented:
An important piece of feedback I've heard from users is:

> _to me the confusion started with finding the tools (I don't think of moderating as a "preference")_

Given this, I've rebased this with the new icons; Turns out my earlier error was due to getting started vs navigation — I'd forgotten the `iconComponent` in one of those.

Updated Screenshot of changes:

<img width="379" alt="Screenshot 2024-08-22 at 19 35 11" src="https://github.com/user-attachments/assets/226f7021-93db-4449-af06-4bdf9e000dba">

renchap reviewed (approved):
Thanks for this. It took us some time, but let's merge.

The potential issue here is making the main menu too big for mods/admins on some smaller-sized screens, but if this is a problem we can probably find a more compact way of displaying those.
